packages/@react-aria/datepicker/src/useDateSegment.ts

```diff
@@ -258,7 +258,7 @@ export function useDateSegment(segment: DateSegment, state: DateFieldState, ref:
 
   let onFocus = () => {
     enteredKeys.current = '';
-    scrollIntoView(getScrollParent(ref.current) as HTMLElement, ref.current);
+    scrollIntoViewFully(ref.current, {block: 'center', inline: 'center'});
```

Comment:
Right now this is the only non `block: nearest` scrollIntoViewFully call, but it brings the experience of keyboard navigating through DateSegments in Safari in line with other browsers. If the original intention was to have this behave like `block: nearest` (which I suspect to be the case with how `scrollIntoView` works), then I'd be happy to adjust. I do think the experience with `{block: 'center', inline: 'center'}` here is nice though since the whole DateField is brought into view and the component is small enough that there won't be weird scroll jumps like in Table when using block: center